text-input.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: page
+title: Text Input
+sidebar: home_sidebar
+permalink: /text-input/
+---
+
+* TOC Placeholder
+{:toc}
+
+## Introduction
+
+This page describe how to set up basic text input for Flutter applications.
+
+The
+[`Input`](https://docs.flutter.io/flutter/material/Input-class.html)
+widget under
+[Material](https://docs.flutter.io/flutter/material/material-library.html )
+satisfies most text input use cases. The `Input` widget implements material
+design's style. If you want an input widget that does not use material design,
+see
+[`RawInputLine`](https://docs.flutter.io/flutter/widgets/RawInputLine-class.html)
+
+
+
+## Retrieving User Input
+
+Assuming that the input is already in focus, the basic data flow for retrieving
+user input is:
+
+1. User taps a character in the keyboard.
+2. The
+[`OnChanged`](https://docs.flutter.io/flutter/material/Input/onChanged.html)
+callback is called with the current `value` of the widget
+3. Perform any necessary logic/validation on the current input value
+4. Update the state of the `Input` widget accordingly through
+[`setState()`](https://docs.flutter.io/flutter/widgets/State/setState.html).
+
+For most cases, we recommend a `Input` within a [`StatefulWidget`](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html)
+so you can save and operate on the current state of the input.
+If you are unfamiliar with the `StatefulWidget` and managing state in Flutter,
```

Comment:
drop "the", so it reads "with StatefulWidget".

s/managing.../how to manage state in Flutter